source/dem/dem.cc

```diff
@@ -748,23 +732,40 @@ DEMSolver<dim>::check_load_balance_dynamic()
   return load_balance_step;
 }
 
+template <int dim>
+inline bool
+DEMSolver<dim>::is_contact_search_step()
+{
+  if (parameters.model_parameters.contact_detection_method ==
+      Parameters::Lagrangian::ModelParameters::ContactDetectionMethod::constant)
+    return check_contact_search_step_constant();
+  else if (parameters.model_parameters.contact_detection_method ==
+           Parameters::Lagrangian::ModelParameters::ContactDetectionMethod::
+             dynamic)
+    return check_contact_search_step_dynamic();
+  else
+    {
+      throw std::runtime_error(
+        "Specified contact detection method is not valid");
+      return false;
+    }
+}
+
 template <int dim>
 inline bool
 DEMSolver<dim>::check_contact_search_step_dynamic()
```

Comment:
Isn't this member not necessary? You could just have find_particle_contact_detection_step as a return function, couldn't you?